specs/core/0_fork-choice.md

```diff
@@ -70,11 +70,12 @@ class Target(object):
 @dataclass
 class Store(object):
     blocks: Dict[Hash, BeaconBlock] = field(default_factory=dict)
-    states: Dict[Hash, BeaconState] = field(default_factory=dict)
-    time: int = 0
-    latest_targets: Dict[ValidatorIndex, Target] = field(default_factory=dict)
-    justified_root: Hash = ZERO_HASH
-    finalized_root: Hash = ZERO_HASH
+    block_states: Dict[Hash, BeaconState] = field(default_factory=dict)
+    checkpoint_states: Dict[Checkpoint, BeaconState] = field(default_factory=dict)
```

Comment:
One is looked up by `(block)` (or the state at the slot the block was created). The other is looked up by checkpoints `(block, epoch)`.  Coercing one into the other doesn't really work nicely so we can't just naively squash, but open to ideas.

Reply:
I see, we can't use `checkpoint_states` everywhere because of `assert block.parent_root in store.block_states`. Would it make sense replace `parent_root` with `parent_checkpoint`?

Reply:
No, because when we get a new block, we store the state at the block/slot. Checkpoints are state at a block/epoch-boundary

Reply:
What about adding `parent_slot`?